Dockerfile

```diff
@@ -66,40 +77,65 @@ VOLUME ["/data", "/var/log/nginx", "/var/log/wsgi"]
 # Expose ports. (http, connection to DB, websocket port)
 EXPOSE 80 443 8005
 
+##############################################
+# APPLICATION
+
 # APP_HOME
 ENV APP_HOME /home/app/django
-
-# Add project directory to docker
-ADD ./ /home/app/django
-
 WORKDIR /home/app/django
 
-RUN cat docker/version >> /etc/profile
+# Install Ruby dependencies
+COPY Gemfile ./
+RUN bundle install
 
-# PIP INSTALL APPLICATION
-RUN pip install -r requirements/production.txt && find . -name '*.pyc' -delete
+# Install node dependencies
+COPY package.json package-lock.json ./
+RUN npm install
 
-# Compile assets
-RUN python manage.py builddata constants_json
+# Install front-end dependencies
+COPY .bowerrc bower.json ./
+RUN npm run bower 
 
-RUN bundle install
+# Build front-end assets
+COPY tasks/ ./tasks
+COPY cla_frontend/assets-src ./cla_frontend/assets-src/
+COPY gulpfile.js ./
+RUN npm run build
 
+# Install socket.io application
+COPY cla_socketserver ./cla_socketserver/
+WORKDIR cla_socketserver
 RUN npm install
 
-RUN node_modules/.bin/bower --allow-root install
+# Install python packages
+WORKDIR $APP_HOME
+COPY requirements/ ./requirements
+COPY requirements.txt ./
+RUN pip install -r requirements/production.txt && find . -name '*.pyc' -delete
```

Comment:
🙋‍♂️ It feels that we change the `requirements` files less often than assets, so these steps could be moved to around the Ruby/NPM/bower dependencies.